metrics/internalrequest.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	InternalRequestAttemptConcurrentTotal = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "internal_request_attempt_concurrent_requests",
+			Help: "Total number of concurrent InternalRequest attempts",
+		},
+	)
+
+	InternalRequestAttemptDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "internal_request_attempt_duration_seconds",
+			Help:    "InternalRequest durations from the moment the InternalRequest PipelineRun was created til the InternalRequest is marked as finished",
+			Buckets: []float64{60, 150, 300, 450, 600, 750, 900, 1050, 1200, 1800, 3600},
+		},
+		[]string{"request", "namespace", "reason", "succeeded"},
+	)
+
+	InternalRequestAttemptRunningSeconds = prometheus.NewHistogram(
```

Comment:
We don't need this one. Not much validation happening here. Not interesting at all.

Reply:
@davidmogar correct of I am wrong: This one means the time the internalrequest is created, the Pod is created, bootstraped and then running, hence, the Pipeline is marked as running. Maybe would it be an interesting clue that could help to identify a potential problem?

Reply:
In the release service this is used to see how much we take to validate a Release. In the Internal Services operator, this would be the time until we create the PipelineRun (not running). So that's probably less than a second.